Source/Configuration/RuntimeFileConfigurationProvider.cs

```diff
@@ -0,0 +1,157 @@
+    readonly IFileProvider? _legacyFilesProvider;
+    readonly Func<IConfigurationBuilder, IFileInfo, IConfigurationRoot> _buildConfigurationFromFile;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RuntimeFileConfigurationProvider"/> class.
+    /// </summary>
+    /// <param name="runtimeConfigFile"></param>
+    /// <param name="legacyFilesProvider"></param>
+    /// <param name="buildConfigurationFromFile"></param>
+    public RuntimeFileConfigurationProvider(IFileInfo runtimeConfigFile, IFileProvider? legacyFilesProvider, Func<IConfigurationBuilder, IFileInfo, IConfigurationRoot> buildConfigurationFromFile)
+    {
+        _runtimeConfigFile = runtimeConfigFile;
+        _legacyFilesProvider = legacyFilesProvider;
+        _buildConfigurationFromFile = buildConfigurationFromFile;
+    }
+
+    /// <inheritdoc />
+    public override void Load()
+    {
+        MapRuntimeConfigFile();
+        MapLegacyDolittleFiles();
+    }
+
+    void MapRuntimeConfigFile()
+    {
+        AddAllFromMap(GetData(Constants.DolittleConfigSectionRoot, _buildConfigurationFromFile(new ConfigurationBuilder(), _runtimeConfigFile)));
+    }
+    
+    void MapLegacyDolittleFiles()
+    {
+        foreach (var file in _legacyFilesProvider?.GetDirectoryContents("/") ?? Enumerable.Empty<IFileInfo>())
+        {
+            MapLegacyDolittleFile(Path.GetFileNameWithoutExtension(file.Name), _buildConfigurationFromFile(new ConfigurationBuilder(), file));
+        }
+    }
+    void MapLegacyDolittleFile(string fileNameWithoutExtension, IConfiguration config)
+    {
+        switch (fileNameWithoutExtension)
+        {
+            case "endpoints":
+                MapIntoRoot("endpoints", config);
+                break;
+            case "metrics":
+                MapIntoRoot(ConfigurationPath.Combine("endpoints", "metrics"), config);
+                break;
+            case "platform":
+                MapIntoRoot("platform", config);
+                break;
+            case "microservices":
+                MapIntoRoot("microservices", config);
+                break;
+            case "resources":
+                foreach (var key in Data.Keys.Where(_ => _.StartsWith(Constants.CombineWithDolittleConfigRoot("tenants"), StringComparison.InvariantCulture)))
+                {
+                    Data.Remove(key);
+                }
+                MapResources(config);
+                break;
+            case "event-horizon-consents":
+                MapEventHorizonConsents(config);
+                break;
+        }
+    }
+
+    void AddAllFromMap(IEnumerable<KeyValuePair<string, string>> map)
+    {
+        foreach (var (key, value) in map)
+        {
+            AddOrReplace(key, value);
+        }
+    }
+
+    void AddOrReplace(string key, string value)
+    {
+        if (Data.TryAdd(key, value))
+        {
+            return;
+        }
+        Data.Remove(key);
+        Data.Add(key, value);
+    }
```

Comment:
Wouldn't this work?
```suggestion
    void AddAllFromMap(IEnumerable<KeyValuePair<string, string>> map)
    {
        foreach (var (key, value) in map)
        {
            Data[key] = value;
        }
    }
```
Maybe a rename to SetAllFromMap would also be cleaner then.